Pretty Print JSON

Squab opened the pull request:
- [X] Adds a function in `yaml_util` to return a String of pretty JSON
- [X] Utilizes `pretty_json` in regular requests
- [X] Utilizes `pretty_json` in remembering requests
- [X] Python unit test for `/rawjson`
- [X] Rust unit test for `pretty_json`

TimSimmons committed:
Pretty Print JSON

- [X] Adds a function in `yaml_util` to return a String of pretty JSON
- [X] Utilizes `pretty_json` in regular requests
- [X] Utilizes `pretty_json` in remembering requests
- [X] Python unit test for `/rawjson`
- [X] Rust unit test for `pretty_json`

pglass commented on src/spag/yaml_util.rs:
```diff
@@ -109,3 +110,11 @@ pub fn get_nested_value<'a>(y: &'a Yaml, keys: &[&str]) -> Option<&'a Yaml> {
 }
+
+// If body can be serialized to JSON, return a pretty JSON string, otherwise return original string 
+pub fn pretty_json(resp_output: String) -> String {
```
So if this function takes a string slice here (`resp_output: &str`) instead of a String, then you shouldn't have to make copies before passing strings in. You would just call `pretty_json(&string)` or `pretty_json(string.as_str())` instead.

TimSimmons committed:
Update dependencies and lock docopt_macros

- `docopt_macros` wasn't compiling, once they do a `0.6.70` release, we can go back
to the regular dependency
- New version of `yaml-rust` (wasn't a pinned dependency so it was being used already)

Squab commented on Cargo.toml:
```diff
@@ -8,8 +8,8 @@ name = "noodles"
-docopt = "0.6.66"
-docopt_macros = "0.6.66"
+docopt = "=0.6.67"
+docopt_macros = "=0.6.66"
```
Failed build: https://travis-ci.org/pglbutt/noodles/builds/73696296

Squab commented:
This should be ready to go @pglass

pglass commented on Cargo.toml:
```diff
@@ -8,8 +8,12 @@ name = "noodles"
+
+[dependencies.docopt_macros]
+git = "https://github.com/docopt/docopt.rs.git"
+rev = "ba5cf65299c5bf4a67ae7c8efae2791a19f612c4"
```
This didn't compile for me... 

```
root@dev-big:~/progs/noodles# cargo build
    Updating registry `https://github.com/rust-lang/crates.io-index`
    Updating git repository `https://github.com/docopt/docopt.rs.git`
 Downloading docopt v0.6.69
   Compiling docopt v0.6.69 (https://github.com/docopt/docopt.rs.git?rev=ba5cf65299c5bf4a67ae7c8efae2791a19f612c4#ba5cf652)
   Compiling docopt v0.6.69
   Compiling docopt_macros v0.6.69 (https://github.com/docopt/docopt.rs.git?rev=ba5cf65299c5bf4a67ae7c8efae2791a19f612c4#ba5cf652)
/root/.cargo/git/checkouts/docopt.rs-81b31a115c87e3e0/ba5cf65299c5bf4a67ae7c8efae2791a19f612c4/docopt_macros/src/macro.rs:262:29: 262:37 error: mismatched types:
 expected `syntax::ast::Name`,
    found `&'static str`
(expected struct `syntax::ast::Name`,
    found &-ptr) [E0308]
/root/.cargo/git/checkouts/docopt.rs-81b31a115c87e3e0/ba5cf65299c5bf4a67ae7c8efae2791a19f612c4/docopt_macros/src/macro.rs:262         if deriving.name != "derive" {
                                                                                                                                                          ^~~~~~~~
/root/.cargo/git/checkouts/docopt.rs-81b31a115c87e3e0/ba5cf65299c5bf4a67ae7c8efae2791a19f612c4/docopt_macros/src/macro.rs:262:29: 262:37 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to previous error
Could not compile `docopt_macros`.

To learn more, run the command again with --verbose.
```

Using `docopt_macros = "0.6.69"` worked just fine though.

pglass replied:
Yeah. I tried again too:

```
(.venv)root@dev-big:~/progs/noodles# rm Cargo.lock
(.venv)root@dev-big:~/progs/noodles# cargo clean
(.venv)root@dev-big:~/progs/noodles# cargo build
    Updating registry `https://github.com/rust-lang/crates.io-index`
    Updating git repository `https://github.com/docopt/docopt.rs.git`
   Compiling yaml-rust v0.2.1
   Compiling gcc v0.3.12
   Compiling libc v0.1.8
   Compiling matches v0.1.2
   Compiling strsim v0.3.0
   Compiling rustc-serialize v0.3.15
   Compiling pkg-config v0.3.5
   Compiling regex-syntax v0.2.1
   Compiling log v0.3.1
   Compiling memchr v0.1.3
   Compiling openssl-sys v0.6.4
   Compiling libz-sys v0.1.7
   Compiling curl-sys v0.1.25
   Compiling aho-corasick v0.3.0
   Compiling regex v0.1.41
   Compiling url v0.2.36
   Compiling docopt v0.6.69 (https://github.com/docopt/docopt.rs.git?rev=ba5cf65299c5bf4a67ae7c8efae2791a19f612c4#ba5cf652)
   Compiling docopt v0.6.69
   Compiling docopt_macros v0.6.69 (https://github.com/docopt/docopt.rs.git?rev=ba5cf65299c5bf4a67ae7c8efae2791a19f612c4#ba5cf652)
/root/.cargo/git/checkouts/docopt.rs-81b31a115c87e3e0/ba5cf65299c5bf4a67ae7c8efae2791a19f612c4/docopt_macros/src/macro.rs:262:29: 262:37 error: mismatched types:
 expected `syntax::ast::Name`,
    found `&'static str`
(expected struct `syntax::ast::Name`,
    found &-ptr) [E0308]
/root/.cargo/git/checkouts/docopt.rs-81b31a115c87e3e0/ba5cf65299c5bf4a67ae7c8efae2791a19f612c4/docopt_macros/src/macro.rs:262         if deriving.name != "derive" {
                                                                                                                                                          ^~~~~~~~
/root/.cargo/git/checkouts/docopt.rs-81b31a115c87e3e0/ba5cf65299c5bf4a67ae7c8efae2791a19f612c4/docopt_macros/src/macro.rs:262:29: 262:37 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to previous error
Build failed, waiting for other jobs to finish...
Could not compile `docopt_macros`.

To learn more, run the command again with --verbose.
```

pglass replied:
Okay, nevermind. This works on updated rust.

Squab commented:
@pglass try upgrading your rust. It worked fine for me but didn't work in Travis. Thus the change.

Squab commented:
Oh derp reading on my phone, didn't see that you merged it :)
